doc/source/index.rst

```diff
@@ -85,8 +85,165 @@ should have.
        cluster_arn="arn:aws:ecs:<region>:<acctid>:cluster/<gpuclustername>",
        worker_gpu=1)
 
+Azure
+-----
+
+In order to start using ``dask_cloudprovider.AzureMLCluster`` you need, at a minimum,
+an Azure subscription, an AzureML workspace, and a quota to create your compute target.
+
+Getting started
+^^^^^^^^^^^^^^^
+
+Imports
+~~~~~~~
+
+First, import all necessary modules.
+
+.. code-block:: python
+
+   from azureml.core import Workspace, Environment
+   from azureml.core.conda_dependencies import CondaDependencies
+   from azureml.core.compute import ComputeTarget, AmlCompute
+   
+   from dask_cloudprovider import AzureMLCluster
+
+   import os
+
+Setup
+~~~~~
+
+Next, create the ``Workspace`` object given your AzureML ``Workspace`` parameters. Check
+more in the AzureML documentation for `Workspace <https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.core.workspace.workspace?view=azure-ml-py>`_.
+
+.. code-block:: python
+
+   subscription_id = '<your-subscription-id-here>'
+   resource_group = '<your-resource-group>'
+   workspace_name = '<your-workspace-name>'
+
+   ws = Workspace(
+      workspace_name=workspace_name
+      , subscription_id=subscription_id
+      , resource_group=resource_group
+   )
+
+Configure parameters
+~~~~~~~~~~~~~~~~~~~~
+
+Let's keep everything in one place so it's easy to maintain.
+
+.. code-block:: python
+
+   name = 'dask-azureml'
+
+   ### vnet settings
+   vnet_rg = ws.resource_group
+   vnet_name = 'dask_azureml_vnet'
+   subnet_name = 'default'
+
+   ### azure ml names: ct - compute target, env - environment
+   ct_name  = 'dask-ct'
+   env_name = 'AzureML-Dask-CPU'
+
+If you are running from an AzureML Compute Instance (Jupyter Lab) you should put both,
+the ``ComputeTarget`` (see `Compute Target <https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.core.compute.computetarget?view=azure-ml-py>`_ documentation page) 
+and the Compute Instance on the
+same `virtual network <https://docs.microsoft.com/en-us/azure/virtual-network/virtual-networks-overview>`_.
+
+The ``AzureMLCluster`` class allows you to submit the job from your local machine as well (tested on DOS CLI and bash). In such case it is not necesary to 
+for the Compute Target to be on a virtual network (it is still preferred to hide the cluster in a virtual network). 
+However, you need to provide some administator name (other than ``admin``) and private and public SSH keys.
+
+.. code-block:: python
+
+   ### credentials
+   admin_username = name.split('-')[0]   ### dask
+   admin_ssh_key_pub = '<path-to-public-key>'
+   admin_ssh_key_priv = '<path-to-private-key>'
+
+The above credentials will be used to create an SSH tunnel between the headnode and your local machine
+so you can communicate with the Dask cluster.
+
+Create Compute Target
+~~~~~~~~~~~~~~~~~~~~~
+
+Next, let's create or retrieve already existing compute target. For a full list of
```

Comment:
Would be nice to have a sentence here like `By default, we will use ``STANDARD_DS13_V2`` VMs because...`